### DISCUSSION TITLE
How to add a custom field to a device component template (i.e. interface template, console port template, etc.)?

### DISCUSSION THREAD

**Discussion by drygdryg** [netbox-community/netbox#15471] [Q&A] [answered]
I would like to add an SNMP ifName to each interface template in a device type. So I need to create a custom field with "Interface template" content type. How to do it?

**Comment by candlerb** [comment, marked as answer]
There are no custom fields for component templates. I think this was discussed fairly recently... #14971

That is: you can add a custom field on "interface", and all interfaces cloned from a template will have that custom field present (but unset). You can't specify *values* for custom fields on the interface template, and have them copied to the clone.

I don't see in principle why it couldn't be done, it just hasn't been.

**Reply by drygdryg** [reply]
I would like my custom field to be stored in the template and never copied to the interfaces instantiated from this template. I need to store some immutable information about the interfaces of all devices of some network device model.

**Reply by candlerb** [reply]
> I would like my custom field to be stored in the template and never copied to the interfaces instantiated from this template

That wouldn't work, because there is no relationship between an interface and the template it was created from (if any). Also, once a device has been instantiated, you can create, delete and rename interfaces on that device. The template is nothing more than a potato print from which the *initial* set of interfaces on a device are copied.
